docs/recipes/DevelopingOnFork.md

```diff
@@ -0,0 +1,201 @@
+        >
+          {isPending ? "Sending..." : "Test Greeting"}
+        </button>
+      </div>
+    </div>
+  );
+};
+```
+</details>
+
+## Implementation Guide
+
+### Step 1: Start the Fork Network
+First, start a local fork of mainnet:
+
+```bash title="terminal"
+# Start fork
+yarn chain --fork-network https://starknet-mainnet.public.blastapi.io/rpc/v0_7
+
+# In a new terminal, deploy contracts
+yarn deploy
+
+# Start frontend on another terminal
+yarn start
+```
+
+### Step 2: Create Network Interaction Component
+Create a new component to interact with your forked network:
+
+```tsx title="components/ForkNetworkInteraction.tsx"
+import { useAccount } from "@starknet-react/core";
+import { useScaffoldContract } from "~~/hooks/scaffold-stark";
+
+export const ForkNetworkInteraction = () => {
+  const { address } = useAccount();
+  const { data: contract } = useScaffoldContract({
+    contractName: "YourContract",
+  });
+  // ... rest of the component
+};
+```
+
+### Step 3: Implement Write Functions
+Add contract interactions using `useScaffoldWriteContract`:
+
+```tsx title="components/ForkNetworkInteraction.tsx"
+const { sendAsync: setGreeting, isPending } = useScaffoldWriteContract({
+  contractName: "YourContract",
+  functionName: "set_greeting",
+  args: ["Testing Fork!"],
+});
+
+const handleSetGreeting = async () => {
+  try {
+    await setGreeting();
+    notification.success("Greeting updated!");
+  } catch (error) {
+    notification.error("Failed to update greeting");
+  }
+};
+```
+
+### Step 4: Add to Your App Page
+Import and use the component in your app:
+
+```tsx title="app/page.tsx"
+import { ForkNetworkInteraction } from "../components/ForkNetworkInteraction";
+
+export default function Home() {
+  return (
+    <div className="container mx-auto">
+      <ForkNetworkInteraction />
+    </div>
+  );
+}
+```
+### Step 5: Testing with Different Wallets
+
+#### Using Burner Wallets
+```tsx title="components/BurnerWalletTest.tsx"
+import { useBurnerWallet } from "~~/hooks/scaffold-stark";
+
+export const BurnerWalletTest = () => {
+  const { connect } = useBurnerWallet();
+  
+  return (
+    <button onClick={connect}>
+      Use Burner Wallet
+    </button>
+  );
+};
+```
+
+#### Connecting External Wallets
+For Argent/Braavos wallets:
+
+```tsx title="components/ExternalWalletConnection.tsx"
+import { useConnect } from "@starknet-react/core";
+
+export const ExternalWalletConnection = () => {
+  const { connect, connectors } = useConnect();
+
+  return (
+    <div>
+      {connectors.map((connector) => (
+        <button
+          key={connector.id}
+          onClick={() => connect({ connector })}
+        >
+          Connect {connector.name}
+        </button>
+      ))}
+    </div>
+  );
+};
+```
+## Testing Different Wallet Types
+
+### 1. Using Burner Wallet
+- Automatically funded with test ETH and STRK to cover gas fees.
+- Perfect for quick testing
+- No external wallet needed
+
+### 2. Using Argent/Braavos
```

Comment:
adding some screenshots would be helpful here, it is important to be able to connect extension wallets and send transactions, on `/debug-ui` is ok

Reply:
argent and braavos